aws/ci_builder.rb

```diff
@@ -0,0 +1,91 @@
+require 'cdo/rake_utils'
+
+module CiBuilder
+  require_relative '../lib/cdo/rake_utils'
+  extend self
+  BUILD_STARTED = 'build-started'.freeze
+
+  def build
+    Dir.chdir(deploy_dir) do
+      return 0 unless build_required?
+
+      FileUtils.touch BUILD_STARTED
+
+      git_update_count = update_repository
+
+      log_build_history(git_update_count)
+
+      display_commit_link
+
+      install_dependencies
+
+      notify_honeybadger
+
+      execute_ci_task
+    end
+  end
+
+  private
+
+  def build_required?
+    RakeUtils.git_updates_available? || File.file?(BUILD_STARTED) || !CDO.daemon
+  end
+
+  def update_repository
+    RakeUtils.git_fetch
+    count = RakeUtils.git_update_count
+    RakeUtils.git_pull if count > 0
+    [1, count].max # I think this is not needed. unless count can be a real number between 0 and 1.
```

Comment:
I think this is just setting `count` to `1` if it was `0`

Reply:
We had a philosophical discussion about this. 

1) We realized we need to refactor this line. Which is simple enough to be included here but fragile enough to warrant its own pull request.


Once this is checked in, the test cases will be added and I should be able to independently test this too